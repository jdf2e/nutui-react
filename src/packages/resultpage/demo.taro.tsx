import React from 'react'
import { ScrollView, View } from '@tarojs/components'
import { useTranslate } from '@/sites/assets/locale/taro'
import Demo1 from './demos/taro/demo1'
import Demo2 from './demos/taro/demo2'
import Demo3 from './demos/taro/demo3'
import Demo4 from './demos/taro/demo4'
import Demo5 from './demos/taro/demo5'

const ResultPageDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基础用法',
      modifyStatus: '修改状态',
      noTitle: '无标题',
      singleButton: '单按钮',
      noButton: '无按钮',
    },
    'zh-TW': {
      basic: '基礎用法',
      modifyStatus: '修改狀態',
      noTitle: '無標題',
      singleButton: '單按鈕',
      noButton: '無按鈕',
    },
    'en-US': {
      basic: 'Basic Usage',
      modifyStatus: 'Modify Status',
      noTitle: 'No Title',
      singleButton: 'Single Button',
      noButton: 'No Button',
    },
  })
  return (
    <>
      <ScrollView className="demo">
        <View className="h2">{translated.basic}</View>
        <Demo1 />
        <View className="h2">{translated.modifyStatus}</View>
        <Demo2 />
        <View className="h2">{translated.noTitle}</View>
        <Demo3 />
        <View className="h2">{translated.singleButton}</View>
        <Demo4 />
        <View className="h2">{translated.noButton}</View>
        <Demo5 />
      </ScrollView>
    </>
  )
}

export default ResultPageDemo
