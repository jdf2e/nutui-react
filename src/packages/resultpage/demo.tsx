import React from 'react'
import { useTranslate } from '@/sites/assets/locale'
import Demo1 from './demos/h5/demo1'
import Demo2 from './demos/h5/demo2'
import Demo3 from './demos/h5/demo3'
import Demo4 from './demos/h5/demo4'
import Demo5 from './demos/h5/demo5'

const ResultPageDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基础用法',
      modifyStatus: '修改状态',
      noTitle: '无标题',
      singleButton: '单按钮',
      noButton: '无按钮',
    },
    'zh-TW': {
      basic: '基礎用法',
      modifyStatus: '修改狀態',
      noTitle: '無標題',
      singleButton: '單按鈕',
      noButton: '無按鈕',
    },
    'en-US': {
      basic: 'Basic Usage',
      modifyStatus: 'Modify Status',
      noTitle: 'No Title',
      singleButton: 'Single Button',
      noButton: 'No Button',
    },
  })
  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Demo1 />
        <h2>{translated.modifyStatus}</h2>
        <Demo2 />
        <h2>{translated.noTitle}</h2>
        <Demo3 />
        <h2>{translated.singleButton}</h2>
        <Demo4 />
        <h2>{translated.noButton}</h2>
        <Demo5 />
      </div>
    </>
  )
}

export default ResultPageDemo
