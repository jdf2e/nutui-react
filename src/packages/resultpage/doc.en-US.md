# ResultPage

## Intro

Feedback the operation results to the user in the form of a page

## Install

```tsx
import { ResultPage } from '@nutui/nutui-react'
```

## Demo

### Basic usage

:::demo

<CodeBlock src='h5/demo1.tsx'></CodeBlock>

:::

### Modify status

:::demo

<CodeBlock src='h5/demo2.tsx'></CodeBlock>

:::

### No Title

:::demo

<CodeBlock src='h5/demo3.tsx'></CodeBlock>

:::

### Single Button

:::demo

<CodeBlock src='h5/demo4.tsx'></CodeBlock>

:::

### No Button

:::demo

<CodeBlock src='h5/demo5.tsx'></CodeBlock>

:::

## ResultPage

### Props

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| title | title | `ReactNode` | `-`|
| description | Description, up to two lines displayed | `ReactNode` | `-` |
| status | Status type | `success` \| `error` \| `warning` \| `info` \| `waiting` | `info` |
| icon | Custom `icon` | `ReactNode` | `-` |
| actions | Actions of operation | `Array` | `[]` |

## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/component/configprovider).

| Name | Description | Default |
| --- | --- | --- |
| \--nutui-resultpage-width | Content area width | `247px` |
| \--nutui-resultpage-icon-size | Width and height of icon | `48px` |
| \--nutui-resultpage-icon-margin-bottom | margin-bottom value of the icon | `16px` |
| \--resultpage-title-margin-bottom | The margin-top value of the title | `9px` |
| \--nutui-resultpage-title-font-size | The font size of the title | `$font-size-xl` |
| \---nutui-resultpage-title-color | The text color of the title | `$color-title` |
| \--nutui-resultpage-description-font-size | Describe the font size | `$font-size-base` |
| \--nutui-resultpage-description-color | Describe the text color | `$color-text` |
| \--nutui-resultpage-description-line-height | Describe the line height | `20px` |
| \--nutui-resultpage-actions-margin-topt | The margin-top value of the operation area | `21px` |