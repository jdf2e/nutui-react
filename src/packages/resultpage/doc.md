# ResultPage组件

## 介绍

以页面的形式向用户反馈操作结果

## 安装

```tsx
import { ResultPage } from '@nutui/nutui-react'
```

## 代码演示

### 基础用法

:::demo

<CodeBlock src='h5/demo1.tsx'></CodeBlock>

:::

### 修改状态

:::demo

<CodeBlock src='h5/demo2.tsx'></CodeBlock>

:::

### 无标题

:::demo

<CodeBlock src='h5/demo3.tsx'></CodeBlock>

:::

### 单按钮

:::demo

<CodeBlock src='h5/demo4.tsx'></CodeBlock>

:::

### 无按钮

:::demo

<CodeBlock src='h5/demo5.tsx'></CodeBlock>

:::

## ResultPage

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| title | 标题 | `ReactNode` | `-`|
| description | 描述，最多展示两行 | `ReactNode` | `-` |
| status | 状态类型 | `success` \| `error` \| `warning` \| `info` \| `waiting` | `info` |
| icon | 自定义 `icon` | `ReactNode` | `-` |
| actions | 可用于处理操作的一组数据 | `Array` | `[]` |

## 主题定制

### 样式变量

组件提供了下列 CSS 变量，可用于自定义样式，使用方法请参考 [ConfigProvider 组件](#/zh-CN/component/configprovider)。

| 名称 | 说明 | 默认值 |
| --- | --- | --- |
| \--nutui-resultpage-width | 内容区域宽度 | `247px` |
| \--nutui-resultpage-icon-size | icon的宽高 | `48px` |
| \--nutui-resultpage-icon-margin-bottom | icon的margin-bottom值 | `16px` |
| \--resultpage-title-margin-bottom | 标题的margin-top值 | `9px` |
| \--nutui-resultpage-title-font-size | 标题的字体大小 | `$font-size-xl` |
| \---nutui-resultpage-title-color | 标题的文字颜色 | `$color-title` |
| \--nutui-resultpage-description-font-size | 描述的字体大小 | `$font-size-base` |
| \--nutui-resultpage-description-color | 描述的文字颜色 | `$color-text` |
| \--nutui-resultpage-description-line-height | 描述的行高 | `20px` |
| \--nutui-resultpage-actions-margin-topt | 操作区域的margin-top值 | `21px` |