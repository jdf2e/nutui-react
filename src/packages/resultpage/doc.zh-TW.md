# ResultPage組件

## 介紹

以頁面的形式向用戶反饋操作結果

## 安裝

```tsx
import { ResultPage } from '@nutui/nutui-react'
```

## 代碼演示

### 基礎用法

:::demo

<CodeBlock src='h5/demo1.tsx'></CodeBlock>

:::

### 修改狀態

:::demo

<CodeBlock src='h5/demo2.tsx'></CodeBlock>

:::

### 無標題

:::demo

<CodeBlock src='h5/demo3.tsx'></CodeBlock>

:::

### 單按鈕

:::demo

<CodeBlock src='h5/demo4.tsx'></CodeBlock>

:::

### 無按鈕

:::demo

<CodeBlock src='h5/demo5.tsx'></CodeBlock>

:::

## ResultPage

### Props

| 屬性 | 說明 | 類型 | 默認值 |
| --- | --- | --- | --- |
| title | 標題 | `ReactNode` | `-`|
| description | 描述，最多展示兩行 | `ReactNode` | `-` |
| status | 狀態類型 | `success` \| `error` \| `warning` \| `info` \| `waiting` | `info` |
| icon | 自定義 `icon` | `ReactNode` | `-` |
| actions | 可用於處理操作的一組數據 | `Array` | `[]` |

## 主題定製

### 樣式變量

組件提供了下列 CSS 變量，可用於自定義樣式，使用方法請參考 [ConfigProvider 組件](#/zh-CN/component/configprovider)。

| 名稱 | 說明 | 默認值 |
| --- | --- | --- |
| \--nutui-resultpage-width | 內容區域寬度 | `247px` |
| \--nutui-resultpage-icon-size | icon的寬高 | `48px` |
| \--nutui-resultpage-icon-margin-bottom | icon的margin-bottom值 | `16px` |
| \--resultpage-title-margin-bottom | 標題的margin-top值 | `9px` |
| \--nutui-resultpage-title-font-size | 標題的字體大小 | `$font-size-xl` |
| \---nutui-resultpage-title-color | 標題的文字顏色 | `$color-title` |
| \--nutui-resultpage-description-font-size | 描述的字體大小 | `$font-size-base` |
| \--nutui-resultpage-description-color | 描述的文字顏色 | `$color-text` |
| \--nutui-resultpage-description-line-height | 描述的行高 | `20px` |
| \--nutui-resultpage-actions-margin-topt | 操作區域的margin-top值 | `21px` |