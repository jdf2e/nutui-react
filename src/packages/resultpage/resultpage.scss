.nut-resultpage {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  &-icon {
    height: $resultpage-icon-size;
    width: $resultpage-icon-size;
    margin-bottom: $resultpage-icon-margin-bottom;
    background-size: 100%;
  }
  &-title {
    width: $resultpage-width;
    margin-bottom: $resultpage-title-margin-bottom;
    font-size: $resultpage-title-font-size;
    color: $resultpage-title-color;
    font-weight: $font-weight-bold;
    text-align: center;
  }
  &-description {
    width: $resultpage-width;
    line-height: $resultpage-description-line-height;
    font-size: $resultpage-description-font-size;
    color: $resultpage-description-color;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
    overflow: hidden;
    word-break: break-all;
  }
  &-actions {
    display: flex;
    flex-direction: row;
    margin-top: $resultpage-actions-margin-top;
  }
  &-action {
    margin-left: 6px;
    margin-right: 6px;
  }
}
