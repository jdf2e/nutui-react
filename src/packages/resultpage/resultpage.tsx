import React, { FunctionComponent, ReactNode, useState } from 'react'
import classNames from 'classnames'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'
import { Button, ButtonFill, ButtonSize, ButtonType } from '../button/button'
import { ResultPageStatus, ResultPageStatusOptions } from './types'

export type ResultPageAction = {
  text: React.ReactNode
  type?: ButtonType
  size?: ButtonSize
  fill?: ButtonFill
  disabled?: boolean
  onClick?: () => () => void
}

export interface ResultPageProps extends BasicComponent {
  title: ReactNode
  description: ReactNode
  icon: ReactNode
  status: ResultPageStatus
  actions: ResultPageAction[]
}
const defaultProps = {
  ...ComponentDefaults,
  title: null,
  description: null,
  icon: '',
  status: 'info',
  actions: [],
} as ResultPageProps
export const ResultPage: FunctionComponent<
  Partial<ResultPageProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const {
    className,
    style,
    title,
    description,
    icon,
    status,
    actions,
    children,
  } = {
    ...defaultProps,
    ...props,
  }

  const classPrefix = `nut-resultpage`
  const cls = classNames(classPrefix, className)

  const defaultStatus: ResultPageStatusOptions = {
    success:
      'https://img11.360buyimg.com/imagetools/jfs/t1/233690/33/17768/1251/66543101F2589003b/f5dcaea8e29c23aa.png',
    error:
      'https://img14.360buyimg.com/imagetools/jfs/t1/224715/38/17932/1426/66543101F1dbc50e8/eaffdb926b00584b.png',
    warning:
      'https://img13.360buyimg.com/imagetools/jfs/t1/200545/24/43507/1219/66543101Fe58cfbe4/c29054555f3bcb4b.png',
    info: 'https://img13.360buyimg.com/imagetools/jfs/t1/199421/5/44230/1320/66543100F23a9466e/0e9eadd5cebf07fb.png',
    waiting:
      'https://img11.360buyimg.com/imagetools/jfs/t1/226266/21/17859/1428/66543101F2dc4c3f3/44e2ae2b51c6e0ed.png',
  }
  const [imgStyle] = useState<any>({
    backgroundImage: `url(${typeof icon === 'string' && icon ? icon : defaultStatus[status]})`,
    backgroundSize: '100% 100%',
  })
  return (
    <div className={cls} style={style}>
      {typeof icon === 'string' ? (
        <div className={`${classPrefix}-icon`} style={imgStyle} />
      ) : (
        icon
      )}
      {typeof title === 'string' && title ? (
        <div className={`${classPrefix}-title`}>{title}</div>
      ) : (
        title
      )}
      {typeof description === 'string' && description ? (
        <div className={`${classPrefix}-description`}>{description}</div>
      ) : (
        description
      )}
      {actions.length ? (
        <div className={`${classPrefix}-actions`}>
          {actions.map((action, index) => {
            const { text, ...rest } = action
            return (
              <div className={`${classPrefix}-action`} key={index}>
                <Button {...rest}>{action?.text}</Button>
              </div>
            )
          })}
        </div>
      ) : null}
      {children}
    </div>
  )
}

ResultPage.displayName = 'NutResultPage'
