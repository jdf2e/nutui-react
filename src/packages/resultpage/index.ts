import { ResultPage } from './resultpage'

export type { ResultPageProps, ResultPageAction } from './resultpage'
export type { ResultPageStatusOptions, ResultPageStatus } from './types'
export default ResultPage
