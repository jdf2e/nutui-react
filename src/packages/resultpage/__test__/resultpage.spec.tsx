import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { ResultPage } from '../resultpage'

test('should render description correctly', () => {
  const { container, getByText } = render(
    <ResultPage description="内容描述内容可折行，建议最多不超过两行建议最多不超过两行内容描述内容可折行，建议最多不超过两行建议最多不超过两行" />
  )
  expect(
    getByText(
      '内容描述内容可折行，建议最多不超过两行建议最多不超过两行内容描述内容可折行，建议最多不超过两行建议最多不超过两行'
    )
  ).toBeTruthy()
  expect(container).toMatchSnapshot()
})
test('should render actions correctly', () => {
  const { container, getByTestId } = render(
    <ResultPage
      title="信息补充"
      description="暂无数据"
      actions={[{ text: '操作1' }, { text: '操作2' }]}
    />
  )
  expect(container.querySelector('.nut-resultpage-actions')).toBeTruthy()
  expect(
    container.querySelectorAll('.nut-resultpage-actions .nut-button').length
  ).toEqual(2)
})

test('should render custom icon correctly', () => {
  const { getByTestId } = render(
    <ResultPage
      description="测试"
      icon={
        <img
          src="https://static-ftcms.jd.com/p/files/61a9e3313985005b3958672e.png"
          alt=""
          data-testid="testImg"
        />
      }
    />
  )
  expect(getByTestId('testImg')).toHaveAttribute(
    'src',
    'https://static-ftcms.jd.com/p/files/61a9e3313985005b3958672e.png'
  )
})
