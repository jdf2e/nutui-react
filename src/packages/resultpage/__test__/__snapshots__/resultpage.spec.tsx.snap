// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render description correctly 1`] = `
<div>
  <div
    class="nut-resultpage"
  >
    <div
      class="nut-resultpage-icon"
      style="background-image: url("https://img13.360buyimg.com/imagetools/jfs/t1/199421/5/44230/1320/66543100F23a9466e/0e9eadd5cebf07fb.png"); background-size: 100% 100%;"
    />
    <div
      class="nut-resultpage-description"
    >
      内容描述内容可折行，建议最多不超过两行建议最多不超过两行内容描述内容可折行，建议最多不超过两行建议最多不超过两行
    </div>
  </div>
</div>
`;
